src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Lowering.LocalRewriter
+{
+    internal class LocalUsingVarRewriter : BoundTreeRewriterWithStackGuard
+    {
+        public static BoundNode Rewrite(BoundStatement statement)
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
+        {
+            ImmutableArray<BoundStatement> statements = (ImmutableArray<BoundStatement>)this.VisitList(node.Statements);
+
+            for (int i = 0; i < statements.Length; i++)
+            {
+                if (statements[i] is BoundLocalDeclaration localDeclaration)
+                {
+                    if (localDeclaration.LocalSymbol.IsUsing)
+                    {
+                        ImmutableArray<LocalSymbol> locals = ImmutableArray.Create<LocalSymbol>(localDeclaration.LocalSymbol);
+
+                        List<BoundStatement> precedingStatements = new List<BoundStatement>();
```

Comment:
Instead of a `List<BoundStatement>` we should use `ArrayBuilder<BoundStatement>.GetInstance()`.